services/core-web/src/components/noticeOfWork/applications/NOWSideMenu.js

```diff
@@ -10,51 +10,117 @@ const { Link } = Anchor;
  */
 
 const propTypes = {
+  history: PropTypes.shape({
+    push: PropTypes.func,
+    replace: PropTypes.func,
+    action: PropTypes.string,
+    location: PropTypes.shape({ hash: PropTypes.string }),
+  }).isRequired,
   location: PropTypes.shape({ hash: PropTypes.string }).isRequired,
-  history: PropTypes.shape({ push: PropTypes.func }).isRequired,
   match: PropTypes.shape({
     params: {
       id: PropTypes.string,
     },
   }).isRequired,
-  route: PropTypes.shape({ route: PropTypes.string, hashRoute: PropTypes.func }).isRequired,
+  route: PropTypes.shape({ hashRoute: PropTypes.func }).isRequired,
 };
 
-export const NOWSideMenu = (props) => {
-  const onChange = (link) => {
-    const { id } = props.match.params;
-    return props.history.push(props.route.hashRoute(id, link));
+export class NOWSideMenu extends Component {
+  static urlRoute = undefined;
+
+  componentDidMount() {
+    // If the user loads the page with a hash in the URL, start them off at the corresponding NoW section.
+    // Note: Because Keycloak authorization adds params to the URL when it redirects, props.location.hash
+    // will be contaminated with extra params that we don't need. All we want is the hash that corresponds
+    // to the NoW section, so we must parse it out. If the hash is "#state", we must ignore it (see example).
+    // For example: #blasting&state=bd74ea1c-09e5-4d7e-810f-d3558969293a&session_state=1c577088-15a8-4ae2-...
+    let link =
+      this.props.location &&
+      this.props.location.hash &&
+      !this.props.location.hash.startsWith("#state")
+        ? this.props.location.hash
+        : undefined;
+
+    if (!link) {
+      return;
+    }
+
+    // Extracts "#blasting" from "#blasting&state=bd74ea1c-09e5-4d7e-810f-d...", for example.
+    link = link.substr(0, link.indexOf("&"));
```

Comment:
I looked into using the "query-string" module but it doesn't provide a cleaner method of extracting the NoW URL fragment (if present).
E.g., for the URL "http://localhost:3000/dashboard/notice-of-work/application/7e6e23cf-0d98-45e2-a059-739d345aeed8/view#underground-exploration", the following code returns the below object (since Keycloak authorization redirects and appends various params to the URL).

`const params = queryString.parse(this.props.location.hash);  
console.log("params:", params);`

`{  
code: "777ef559-1cc2-4dbb-a646-488450d77a2d.e2cef18a-5b94…6b1096fbf63c.ba58208e-e55c-417b-b845-bd472b79c6da",  
session_state: "e2cef18a-5b94-4b69-8142-6b1096fbf63c",  
state: "fedeef5e-6487-4f7b-80df-310db80e69b3",  
underground-exploration: null  
}`